code/modules/research/xenobiology/crossbreeding/loyal.dm

```diff
@@ -0,0 +1,745 @@
+
+/datum/component/loyal_effect/blue
+	prefix = "clean"
+
+/datum/component/loyal_effect/blue/Initialize(obj/item/attached_item)
+	. = ..()
+	START_PROCESSING(SSobj,src)
+
+/datum/component/loyal_effect/blue/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	return ..()
+
+/datum/component/loyal_effect/blue/process()
+	SEND_SIGNAL(attached, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_GOD) //The CLEANEST
+
+/obj/item/slimecross/loyal/metal
+	colour = "metal"
+	effect_desc = "Makes an item more resistance to brute attacks."
+	loyaleffect = /datum/component/loyal_effect/metal
+
+/datum/component/loyal_effect/metal
+	prefix = "metallic"
+
+/datum/component/loyal_effect/metal/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.armor.attachArmor(list("melee" = 10, "bullet" = 5, "bomb" = 5))
+
+/obj/item/slimecross/loyal/yellow
+	colour = "yellow"
+	effect_desc = "Makes an item glow."
+	loyaleffect = /datum/component/loyal_effect/yellow
+
+/datum/component/loyal_effect/yellow
+	prefix = "glowing"
+
+/datum/component/loyal_effect/yellow/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.set_light(l_range = 5, l_power = 1) //A bit wider than a flashlight
+
+/obj/item/slimecross/loyal/darkpurple
+	colour = "dark purple"
+	effect_desc = "Makes an item combustible and explosive."
+	loyaleffect = /datum/component/loyal_effect/darkpurple
+
+/datum/component/loyal_effect/darkpurple
+	prefix = "combustible"
+
+/obj/item/slimecross/loyal/darkblue
+	colour = "dark blue"
+	effect_desc = "Makes an item fireproof and colder. If clothing, it will cool down whoever wears it. Hitting someone with this will make them quite chilly."
+	loyaleffect = /datum/component/loyal_effect/darkblue
+
+/datum/component/loyal_effect/darkblue
+	prefix = "cryonic"
+	var/mob/living/carbon/attached_user
+
+/datum/component/loyal_effect/darkblue/Initialize(obj/item/attached_item)
+	. = ..()
+	attached_item.resistance_flags |= FIRE_PROOF
+	RegisterSignal(attached_item,COMSIG_ITEM_ATTACK,.proc/cryo_attack)
+	if(istype(attached_item, /obj/item/clothing))
+		RegisterSignal(attached_item, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/equippedChanged)
+		START_PROCESSING(SSobj,src)
+
+/datum/component/loyal_effect/darkblue/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	return ..()
+
+/datum/component/loyal_effect/darkblue/process()
+	if(!attached_user)
+		return
+	if(attached_user.bodytemperature > BODYTEMP_NORMAL) //Stolen from leporazine code
+		attached_user.adjust_bodytemperature(-40 * TEMPERATURE_DAMAGE_COEFFICIENT, BODYTEMP_NORMAL)
+
+/datum/component/loyal_effect/darkblue/proc/cryo_attack(datum/source, mob/living/carbon/C, mob/user)
+	. = ..()
+	C.adjust_bodytemperature(150 - C.bodytemperature) //Stolen from cryo projectile code
+
+/datum/component/loyal_effect/darkblue/proc/equippedChanged(datum/source, mob/living/carbon/user, slot)
+	if(istype(user))
+		attached_user = user
+	else
+		attached_user = null
+
+/obj/item/slimecross/loyal/silver
```

Comment:
Wondering how to make this one work. Maybe something similar to moth clothe eating code?